### PR TITLE
kernel: sched: Use ticks as time unit in time slicing.

### LINKED ISSUE #8896: Failed test: kernel.sched.slice_reset and kernel.sched.slice_scheduling (tests/kernel/sched/schedule_api/kernel.sched) on nrf51/nrf52
```
***** delaying boot 1000ms (per build configuration) *****
 ***** Booting Zephyr OS 1.12.99 (delayed boot 1000ms) *****
 Running test suite threads_scheduling
 ===================================================================
 starting test - test_priority_cooperative
 PASS - test_priority_cooperative
 ===================================================================
 starting test - test_priority_preemptible
 PASS - test_priority_preemptible
 ===================================================================
 starting test - test_yield_cooperative
 PASS - test_yield_cooperative
 ===================================================================
 starting test - test_sleep_cooperative
 PASS - test_sleep_cooperative
 ===================================================================
 starting test - test_sleep_wakeup_preemptible
 PASS - test_sleep_wakeup_preemptible
 ===================================================================
 starting test - test_pending_thread_wakeup
 PASS - test_pending_thread_wakeup
 ===================================================================
 starting test - test_time_slicing_preemptible
 PASS - test_time_slicing_preemptible
 ===================================================================
 starting test - test_time_slicing_disable_preemptible
 PASS - test_time_slicing_disable_preemptible
 ===================================================================
 starting test - test_lock_preemptible
 PASS - test_lock_preemptible
 ===================================================================
 starting test - test_unlock_preemptible
 PASS - test_unlock_preemptible
 ===================================================================
 starting test - test_sched_is_preempt_thread
 PASS - test_sched_is_preempt_thread
 ===================================================================
 starting test - test_slice_reset

     Assertion failed at zephyr.git/tests/kernel/sched/schedule_api/src/test_sched_timeslice_reset.c:39: thread_tslice: t <= expected_slice[thread_idx] is false

 FAIL - test_slice_reset
 ===================================================================
 starting test - test_slice_scheduling
 A
     Assertion failed at zephyr.git/tests/kernel/sched/schedule_api/src/test_slice_scheduling.c:48: thread_tslice: ((tdelta <= SLICE_SIZE) && ((int)p1 == thread_idx)) is false

 B
     Assertion failed at zephyr.git/tests/kernel/sched/schedule_api/src/test_slice_scheduling.c:48: thread_tslice: ((tdelta <= SLICE_SIZE) && ((int)p1 == thread_idx)) is false



     Assertion failed at zephyr.git/tests/kernel/sched/schedule_api/src/test_slice_scheduling.c:48: thread_tslice: ((tdelta <= SLICE_SIZE) && ((int)p1 == thread_idx)) is false

 FAIL - test_slice_scheduling
 ===================================================================
 starting test - test_priority_scheduling
 PASS - test_priority_scheduling
 ===================================================================
 ===================================================================
 RunID: ci-180710-1801-1919:h6ps
 PROJECT EXECUTION FAILED

```

### PR DESCRIPTION
**Pull request by pizi-nordic**
The time slicing settings was kept in milliseconds while all related operations was based on ticks. Continuous back and forth conversion between ticks and milliseconds introduced an accumulating error due to rounding in _ms_to_ticks() and __ticks_to_ms(). As result configured time slice duration was not achieved.

This PR removes excessive ticks <-> ms conversion by using ticks as time unit for all operations related to time slicing.

Also, it fixes #8896, #8897 and #9310

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]

File: kernel/sys_clock.c [hunk 1]
```diff
@@ -263,8 +263,8 @@ static inline void handle_timeouts(s32_t ticks)
 
 #ifdef CONFIG_TIMESLICING
 s32_t _time_slice_elapsed;
-s32_t _time_slice_duration = CONFIG_TIMESLICE_SIZE;
-int  _time_slice_prio_ceiling = CONFIG_TIMESLICE_PRIORITY;
+s32_t _time_slice_duration;
```
Comment (ioannisg) [inline kernel/sys_clock.c]: duration and ceiling are now initialized to 0, right?

Comment (ioannisg) [reply kernel/sys_clock.c]: Or, actually, they are initialized by `k_sched_time_slice_set` in` _sched_init`, right?

Comment (pizi-nordic) [reply kernel/sys_clock.c]: Yes

**Review by ioannisg — APPROVED** [review]
Looks OK to me.

**Pull request merged**